config/v3_4_experimental/translate/translate.go

```diff
@@ -27,9 +28,33 @@ func translateIgnition(old old_types.Ignition) (ret types.Ignition) {
 	return
 }
 
+func translateFileEmbedded1(old old_types.FileEmbedded1) (ret types.FileEmbedded1) {
+	tr := translate.NewTranslator()
+	tr.Translate(&old.Append, &ret.Append)
+	tr.Translate(&old.Contents, &ret.Contents)
+	if util.CheckForSpecialModeBits(old.Mode) {
+		ret.Mode = util.IntToPtr(*old.Mode & 0777)
+	} else {
+		tr.Translate(&old.Mode, &ret.Mode)
```

Comment:
We can just unconditionally mask out the relevant bits, even if they're zero.

Reply:
ok, so the reason behind this `condition` is to address the `nil pointer deference` error

Reply:
https://jenkins-coreos-ci.apps.ocp.fedoraproject.org/blue/organizations/jenkins/ignition/detail/PR-1325/5/pipeline